plugin/audit_log/audit_log.c

```diff
@@ -798,11 +839,14 @@ void audit_log_notify(MYSQL_THD thd __attribute__((unused)),
           strncmp(event_general->general_command, "Quit", 4) == 0)
         break;
 
+      if (local->skip_logging)
```

Comment:
Probably we would want to skip connect/disconnect events for filtered users but not change user events to/from unfiltered/filtered to filtered/unfiltered (this would be a good addition to the testcase too)